File: SlidingWindow/FindKBeautyNumber.java
package SlidingWindow;

public class FindKBeautyNumber {
    
    public static int divisorSubString(int num, int k) {
        String numStr = String.valueOf(num);
        int count = 0;
        int n = numStr.length();

        // Sliding window will check every subString on length k
        for (int i = 0; i <= n - k; i++) {
            String subString = numStr.substring(i, i+k);
            int divisor = Integer.parseInt(subString);
            System.out.println("Divisor : " + divisor);

            if (divisor != 0 && num % divisor == 0) {
                count++;
            }
        }

        return count;
    }
}

File: SlidingWindow/FindKBeautyNumber.py
class FindKBeautyNumber:
    def divisorSubString(self, num: int, k: int) -> int:
        numstr = str(num)
        count = 0
        n = len(numstr)

        # Sliding window of size k
        for i in range(0, n - k + 1):
            sub_str = numstr[i:i+k]
            divisor = int(sub_str)
            print(f"Divisor : {divisor}")

            # check if the divisor is not zero and divides the number
            if divisor != 0 and num % divisor == 0:
                count += 1

        return count

def main():
    sol = FindKBeautyNumber()
    num = 240
    k = 2
    res = sol.divisorSubString(num, k)
    print(res)      

if __name__ == "__main__":  
    main()